### DISCUSSION TITLE
How to use q-form with nicegui

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3348] [Q&A] [answered]
### Question

How to use q-form with nicegui? When we place elements within a q-form, what changes will it  lead to , it seems that the interface will refresh each time an input is completed.
How can we correctly use q-form if we don't want the value of the input to take effect immediately, but rather confirm it through a button?

```py
with ui.element('q-form'):
    i=ui.input('hello')
    i2=ui.input('hello2')
    ui.button('submit or confirm').props('type="submit"').on_click(lambda: print(i.value))   
```

**Comment by falkoschindler** [comment]
@v479038280 What exactly is the problem with `i` and `i2` changing values before pressing the submit button? What do you mean with "take effect immediately"?

**Reply by python-and-novella** [reply]
@v479038280  'lazy-rules' of input？

**Reply by v479038280** [reply]
Usually we bind the value of one element to another element , the bind take effect immediately , what if we want get the value of several ui.input() ,would the q-form be the right way ? with the submit button we confirm the value of these ui.input(), not lazy-rules.

**Reply by python-and-novella** [reply, marked as answer]
without submit or q-form, you can do it by python normally as you said.
